tests/substrate_tests/arrays.rs

```diff
@@ -743,7 +743,7 @@ fn dynamic_array_push() {
 
         contract foo {
             function test() public {
-                int[] bar = new int[](1);
+                int[] bar = (new int[])(1);
```

Comment:
Why are you adding parenthesis here? Shouldn't we keep testing the old syntax too?

Reply:
There are many tests for the existing syntax already. I wanted to add tests for the other syntax too.

There are a few in the same file, here for example.

https://github.com/hyperledger-labs/solang/blob/1c81845dda633e7e749978c57bdc99b9b105d65d/tests/substrate_tests/arrays.rs#L556